src/vscode-dts/vscode.d.ts

```diff
@@ -19654,6 +19737,306 @@ declare module 'vscode' {
 		 */
 		canSendRequest(chat: LanguageModelChat): boolean | undefined;
 	}
+
+	/**
+	 * A tool that is available to the language model via {@link LanguageModelChatRequestOptions}. A language model uses all the
+	 * properties of this interface to decide which tool to call, and how to call it.
+	 */
+	export interface LanguageModelChatTool {
+		/**
+		 * The name of the tool.
+		 */
+		name: string;
+
+		/**
+		 * The description of the tool.
+		 */
+		description: string;
+
+		/**
+		 * A JSON schema for the parameters this tool accepts.
+		 */
+		parametersSchema?: object;
+	}
+
+	/**
+	 * A tool-calling mode for the language model to use.
+	 */
+	export enum LanguageModelChatToolMode {
+		/**
+		 * The language model can choose to call a tool or generate a message. Is the default.
+		 */
+		Auto = 1,
+
+		/**
+		 * The language model must call one of the provided tools. Note- some models only support a single tool when using this
+		 * mode.
+		 */
+		Required = 2
+	}
+
+	/**
+	 * A language model response part indicating a tool call, returned from a {@link LanguageModelChatResponse}, and also can be
+	 * included as a content part on a {@link LanguageModelChatMessage}, to represent a previous tool call in a chat request.
+	 */
+	export class LanguageModelToolCallPart {
+		/**
+		 * The name of the tool to call.
+		 */
+		name: string;
+
+		/**
+		 * The ID of the tool call. This is a unique identifier for the tool call within the chat request.
+		 */
+		callId: string;
+
+		/**
+		 * The parameters with which to call the tool.
+		 */
+		parameters: object;
+
+		/**
+		 * Create a new LanguageModelToolCallPart.
+		 */
+		constructor(name: string, callId: string, parameters: object);
```

Comment:
put `callId` first?

Reply:
Tricky because I am trying to do this without breaking changes, and this is the one thing I can't avoid. I will do this after testing then make sure it gets adopted by each extension.